src/core/core.interaction.js

```diff
@@ -25,15 +25,14 @@ function getRelativePosition(e, chart) {
  * @param {function} handler - the callback to execute for each visible item
  */
 function parseVisibleItems(chart, handler) {
-	var metasets = chart._getSortedVisibleDatasetMetas();
-	var metadata, i, j, ilen, jlen, element;
-
-	for (i = 0, ilen = metasets.length; i < ilen; ++i) {
-		metadata = metasets[i].data;
-		for (j = 0, jlen = metadata.length; j < jlen; ++j) {
-			element = metadata[j];
-			if (!element._view.skip) {
-				handler(element, i, j);
+	const metasets = chart._getSortedVisibleDatasetMetas();
+
+	for (let i = 0, ilen = metasets.length; i < ilen; ++i) {
```

Comment:
I don't think it's necessarily worth changing, but I saw something about how `let` inside a `for` loop is less performant because it's creating a new variable with each iteration. I don't know how much of a real issue that is though

Reply:
it sounds like something that will get optimized away (if not yet then eventually) since it's trivial to test even from an AST whether `i` needs to ever persist uniquely beyond the current iteration's lifetime - if not it can be treated the same as a var.

Reply:
not sure if you removed the `const` due to this comment. I wasn't even thinking about those was was actually talking about the `let i = 0`. I'm fine with it either way though